### PR TITLE
Make sha2 asm as feature

### PR DESCRIPTION
**Pull request by TimeEngineer**

### CODE REVIEW

File: oxidation/crates/parsec_api_crypto/Cargo.toml [hunk 1]
```diff
@@ -62,6 +62,7 @@ use-rustcrypto = [
     "rand_07",
     "rand_08",
 ]
+asm = ["sha2?/asm"]
```
Comment (touilleMan) [inline oxidation/crates/parsec_api_crypto/Cargo.toml:65]: what is this syntax with a interrogation point ?

And I think you should add a comment to explain why this is needed

Comment (FirelightFlagboy) [reply oxidation/crates/parsec_api_crypto/Cargo.toml:67]: why not enabling the feature `sha2/asm` when the `target_arch` is not `wasm32` ?

Comment (touilleMan) [reply oxidation/crates/parsec_api_crypto/Cargo.toml:67]: @FirelightFlagboy the issue is not related to wasm32: the compilation seems to fail with msvc (see https://github.com/RustCrypto/hashes/issues/315)

but we could disable this feature for windows instead

**Review by FirelightFlagboy — APPROVED** [review]

File: oxidation/crates/parsec_api_crypto/Cargo.toml (hunk 1 above)
Comment (TimeEngineer) [reply oxidation/crates/parsec_api_crypto/Cargo.toml:67]: In the doubt, we can leave this, we don't know about others platforms

**Pull request merged**